### PR TITLE
Update ZipJs from 2.4.0/2.5.0 to 3.1.5

### PR DESCRIPTION
**Pull request by kestred**
Resolves https://github.com/calvinmetcalf/shapefile-js/issues/88

### CODE REVIEW

**Comment by calvinmetcalf** [comment]
hey sorry was on vacation, will look into this soon

**Comment by dteske25** [comment]
@calvinmetcalf Sorry to bug you, but I was wondering if you had a chance to check this over yet. I'm having some performance issues with large zips, and was wondering if this update might solve them.

File: lib/index.js [hunk 1]
```diff
@@ -59,52 +59,73 @@ shp.combine = function(arr) {
...
-  var zip = unzip(buffer);
-  var names = [];
-  whiteList = whiteList || [];
-  for (key in zip) {
-    if (key.indexOf('__MACOSX') !== -1) {
-      continue;
+  return unzip(buffer).then(function(zip) {
+    var names = [];
+    whiteList = whiteList || [];
+    if (Object.keys) {
```
Comment (calvinmetcalf) [inline lib/index.js]: if you want to get rid of the for in loop, we can talk about that, but lets put that into a different pull

File: lib/index.js [hunk 2]
```diff
@@ -134,7 +155,9 @@ shp.getShapefile = function(base, whiteList) {
     }
   } else {
     return new Promise(function(resolve) {
-      resolve(shp.parseZip(base));
+      shp.parseZip(base).then(function(geojson) {
```
Comment (calvinmetcalf) [inline lib/index.js]: we can just return shp.parseZip(base) here

**Comment by kestred** [comment]
Original fork repository was archived, see https://github.com/calvinmetcalf/shapefile-js/pull/94 with changes

**Pull request closed without merging**